nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroFactoryContext.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.avro.Schema;
+
+/**
+ * An AvroFactoryContext represents a model context in which avro schemas are cached and reused depending on their
+ * qualified name.<br>
+ * <br>
+ * In the avro schema, the first occurence of the record is fully described, the following occurences are name
+ * references.<br>
+ *
+ * @since 10.2
+ */
+public class AvroFactoryContext {
+
+    private static final AvroFactory<Object> NULL = new AvroFactory<Object>(null) {
+
+        @Override
+        public Schema createSchema(Object input) {
+            return null;
+        }
+
+        @Override
+        public String getName(Object input) {
+            return null;
+        }
+    };
+
+    protected Collection<AvroReplacementDescriptor> replacements;
+
+    protected Map<Class<?>, AvroFactory<?>> factories = new HashMap<>();
+
+    protected Map<String, Schema> cache = new HashMap<>();
```

Comment:
I would go for `computedSchemas` because it's not really a cache here (no eviction).